MatrixKit/Models/Room/MXKRoomDataSource.m

```diff
@@ -2177,23 +2177,26 @@ - (void)handleUnsentMessages
 {
     // Add the unsent messages at the end of the conversation
     NSArray<MXEvent*>* outgoingMessages = _room.outgoingMessages;
-    BOOL shouldProcessQueuedEvents = NO;
     
-    for (NSInteger index = 0; index < outgoingMessages.count; index++)
-    {
-        MXEvent *outgoingMessage = [outgoingMessages objectAtIndex:index];
+    [self.mxSession decryptEvents:outgoingMessages inTimeline:nil onComplete:^(NSArray<MXEvent *> *failedEvents) {
+        BOOL shouldProcessQueuedEvents = NO;
         
-        if (outgoingMessage.sentState != MXEventSentStateSent)
+        for (NSInteger index = 0; index < outgoingMessages.count; index++)
```

Comment:
Remark: Prefer `for in` loop and a separate increment var for performances. It's ok to keep things as is for `outgoingMessages`

Reply:
yep. I did not touch this X years old code.